lib/common/quotation_service.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/material.dart';
+import 'package:localization/localization.dart';
+import 'package:tamini_app/components/quotations/quotations_model.dart';
+
+class QuotationService {
+  static Future<void> updateQuotation(context, String requestId, Map<String, dynamic> updates, String massage) async {
+    try {
+      await FirebaseFirestore.instance.collection('quotations').doc(requestId).update(updates);
+
+      await showSnackbar(context, massage.i18n());
+    } catch (e) {
+      await showSnackbar(context, 'Error: $e');
+    }
+  }
+
+  static Future<void> showSnackbar(context, String message) async {
+    final snackBar = SnackBar(content: Text(message));
+    ScaffoldMessenger.of(context).showSnackBar(snackBar);
+  }
+
+  static listenToUserQuotations(uid, updateUI) {
+    final collection =
+        FirebaseFirestore.instance.collection('quotations').where('userId', isEqualTo: '$uid').snapshots();
+
+    collection.listen((snapshot) {
+      List<Quotations> newList = [];
+
+      for (final doc in snapshot.docs) {
+        final quotation = Quotations.fromMap(doc.data());
+        newList.add(quotation);
+      }
+      newList.sort((b, a) => a.requestId.compareTo(b.requestId));
+      updateUI(newList);
+    });
+  }
+
+  static listenToOwnerQuotations(updateUI) {
+    final collection = FirebaseFirestore.instance.collection('quotations').snapshots();
+    collection.listen((snapshot) {
+      List<Quotations> newList = [];
+      for (final doc in snapshot.docs) {
+        final quotation = Quotations.fromMap(doc.data());
+        newList.add(quotation);
+      }
+      newList.sort((b, a) => a.requestId.compareTo(b.requestId));
+      updateUI(newList);
+    });
+  }
+
+  static Future<void> requestQuotation(
+    context,
+    String nationalId,
+    String birthDate,
+    String carSerialNumber,
+    String uid,
+    String phoneNumber,
+    String message,
+  ) async {
+    final DateTime requestDate = DateTime.now();
+
+    try {
+      DocumentSnapshot metadata = await FirebaseFirestore.instance.collection('metadata').doc('requestId').get();
+      Map<String, dynamic>? data = metadata.data() as Map<String, dynamic>?;
+      int currentRequestId;
+      if (data != null && data.containsKey('currentRequestId')) {
+        currentRequestId = data['currentRequestId'];
+      } else {
+        currentRequestId = 1000;
+      }
+
+      int newRequestId = currentRequestId + 1;
+
+      await FirebaseFirestore.instance.collection('metadata').doc('requestId').set({
+        'currentRequestId': newRequestId,
+      });
```

Comment:
instead of having 2 different http requests, you might get the last id and update it within the same request. I know something similar doable in mongodb. Check if firebase support update document.

example;
```suggestion
      /// below code is just for explanation, not valid.
      FirebaseDocument doc = await FirebaseFirestore.instance.collection('metadata').doc('requestId');
      firebase.update("requestId", doc.requestId.value + 1);
```